NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -256,3 +415,517 @@ def make_sim_shower(corsika, observer=None, detector=None, station_id=None):
+    """
+    Calculate weights according to the area that one simulated position in readCoreasStation represents.
+    Weights are therefore given in units of area.
+    Note: The volume of a 2d convex hull is the area.
+    
+    Parameters
+    ----------
+    positions : list
+        station position with [x, y, z] on ground
+    zenith : float
+        zenith angle of the shower
+    azimuth : float
+        azimuth angle of the shower
+    site : str
+        site of the simulation, default is 'summit'
+    debug : bool
+        if true, plots are created for debugging
+    
+    Returns
+    -------
+    weights : np.array
+        weights of the observer position
+    """
+
+    import scipy.spatial as spatial
+
+    positions = np.array(positions)
+    cs = coordinatesystems.cstrafo(zenith=zenith, azimuth=azimuth, magnetic_field_vector=None,
+                                                   site=site)
+    x_trafo_from_shower = cs.transform_from_vxB_vxvxB(station_position=np.array([1, 0, 0]))
+    y_trafo_from_shower = cs.transform_from_vxB_vxvxB(station_position=np.array([0, 1, 0]))
+    z_trafo_from_shower = cs.transform_from_vxB_vxvxB(station_position=np.array([0, 0, 1]))
+
+    # voronoi has to be calculated in the shower plane due to symmetry reasons
+    shower = cs.transform_to_vxB_vxvxB(station_position=positions)
+    vor = spatial.Voronoi(shower[:, :2])  # algorithm will find no solution if flat simulation is given in 3d.
+
+    if debug:
+        fig1 = plt.figure(figsize=[12, 4])
+        ax1 = fig1.add_subplot(121)
+        ax2 = fig1.add_subplot(122)
+        spatial.voronoi_plot_2d(vor, ax1)
+        ax1.set_aspect('equal')
+        ax1.set_title('In shower plane, zenith = {:.2f}'.format(zenith / units.degree))
+        ax1.set_xlabel(r'Position in $\vec{v} \times \vec{B}$ - direction [m]')
+        ax1.set_ylabel(r'Position in $\vec{v} \times \vec{v} \times \vec{B}$ - direction [m]')
+
+    weights = np.zeros_like(positions[:, 0])
+    for p in range(0, weights.shape[0]):  # loop over all observer positions
+        vertices_shower_2d = vor.vertices[vor.regions[vor.point_region[p]]]
+
+        # x_vertice_ground = x_trafo_from_shower[0] * x_vertice_shower + y_trafo_from_shower[0] * y_vertice_shower + z_trafo_from_shower[0] * z_vertice_shower
+        # y_vertice_ground = x_trafo_from_shower[1] * x_vertice_shower + y_trafo_from_shower[1] * y_vertice_shower + z_trafo_from_shower[1] * z_vertice_shower
+        # z_vertice_ground = x_trafo_from_shower[2] * x_vertice_shower + y_trafo_from_shower[2] * y_vertice_shower + z_trafo_from_shower[2] * z_vertice_shower
+
+        x_vertice_shower = vertices_shower_2d[:, 0]
+        y_vertice_shower = vertices_shower_2d[:, 1]
+        z_vertice_shower = -(x_trafo_from_shower[2] * x_vertice_shower + y_trafo_from_shower[2] * y_vertice_shower) / z_trafo_from_shower[2]
+
+        vertices_shower_3d = []
```

Comment:
I think this forloop can be replace by a concatenation?

Reply:
yes! I did :)